source/fundamentals/crud/read-operations/sort.txt

```diff
@@ -4,135 +4,396 @@ Sort Results
 
 .. default-domain:: mongodb
 
-Use ``sort()`` to change the order in which read operations return
-documents. The ``sort()`` instance method tells MongoDB to order returned documents by the
-values of one or more fields in a certain direction. To sort returned
-documents by a field in ascending (lowest first) order, use the static method ``Sorts.ascending()``, 
-specifying the field on which to sort. To sort in descending (greatest first) order instead, 
-use the static method ``Sorts.descending()``, specifying the field on which to sort. For example,
-you can sort on ``id`` or ``name`` fields. If you do not specify a sort, MongoDB does not 
-guarantee the order of query results.
-
-Follow the instructions in the examples below to insert data into
-a collection and perform a sort on the results of a query.
-Consider a collection containing documents that describe books. To
-insert this data into a collection, run the following operation:
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, we show you how to **sort** your results.
+
+Sorting orders the documents returned from your query by your specified 
+**sort criteria**. Sort criteria are the rules you give MongoDB that describe
+how you would like your data ordered. Some examples of sort criteria are:
+
+* Smallest number to largest number
+* Earliest time of day to latest time of day
+* Alphabetical order by first name 
+
+You should read this guide if you would like to:
+
+* Perform ascending sorts and descending sorts.
+* Combine sort criteria.
+* Sort on the text score of a
+  :manual:`text search </core/text-search-operators/>`.
+
+.. _sorts-crud-sort-example:
+
+The examples in this guide use a sample collection, ``sort_example``, that
+contains the following documents:
+  
+.. code-block:: json
+  
+   {"_id": 1, "letter": "c", "food": "coffee with milk"}
+   {"_id": 3, "letter": "a", "food": "maple syrup"}
+   {"_id": 4, "letter": "b", "food": "coffee with sugar"}
+   {"_id": 5, "letter": "a", "food": "milk and cookies"}
+   {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+   {"_id": 6, "letter": "c", "food": "maple donut"}
+
+Methods For Sorting
+-------------------
+
+You can sort your results using the  ``sort()`` method of a ``FindIterable``
+instance or by using the  ``Aggregates.sort()`` static factory method in an
+aggregation pipeline. Both of these methods take objects that implement the
+``Bson`` interface and specify sort criteria as arguments. For information
+on the ``Bson`` interface, see our 
+:doc:`guide on Bson </fundamentals/data-formats/document-data-format-bson/>`.   
+
+.. _sorts-crud-find-iterable-example:
+
+You can use the ``sort()`` method of a ``FindIterable`` instance as follows:
+
+.. code-block:: java
+
+   // <MongoCollection setup code here>
+   
+   collection.find().sort(new Document("_id", 1)));
```

Comment:
Suggestion: I think you should mention what 1 and -1 means.  The reader might not make the connection with your equivalent example.

Reply:
Thank you for the suggestion. Your suggestion raises a lot of interesting questions. I am not sure what the right call is here, as we are trying to push users towards the builders style for creating sort specifications. Not sure how in depth we should go into MQL? Should we explain a code snippet that we are actively trying to discourage users from using? Would be curious to see what other people on the team have to say to this point?

@ccho-mongodb, @terakilobyte, any thoughts? Should we explain patterns we are trying to have users avoid?

Reply:
I am trying to figure out what you (Alek) want to do since you're presenting both the MQL here and Sorts builder further down. If you want to keep both, I think Kailie's suggestion is a valid point. You could explain it in as little as a sentence and link back to the server manual for more info.

My preference is not to show both on this page because it extends the amount of explanation needed and can easily be missed because it requires vertical scrolling and is further separated by the Aggregates code example. The only page I might provide this comparison is on the Sort builders page to show equivalency.

Reply:
Thank you for the suggestion. After implementing the rewrite, I agree that the `Sorts` builder page is a more natural location for the equivalency example. Fix: https://github.com/mongodb/docs-java/pull/66/commits/a7c3b22a9f311aba0269920c9763ff42a08ee906